pkg/sql/create.go

```diff
@@ -947,6 +947,12 @@ func resolveFK(
 			}
 		}
 		if !found {
+			// Avoid unexpected index builds from ALTER TABLE ADD CONSTRAINT.
+			if mode == sqlbase.ConstraintValidity_Unvalidated {
+				return errors.Errorf(
+					"foreign key columns %s must be the prefix of an index", colNames(srcCols),
```

Comment:
sure. one thing to note is that a pre-existing index need only have the columns they're trying to use as a *prefix* (since that is good enough for retrieval), unlike the referenced index, which we, and postgres, require be unique, and thus must match the constrained columns exactly. 

Do we feel like trying to clarify this subtly in a error message? or should we just let the docs take over here if someone really wants to get into he details?